### PR TITLE
[processing] support only SAGA LTR

### PR DESCRIPTION
**Pull request by alexbruy**

### CODE REVIEW

**Comment by ghtmtt** [comment]
I'm on a Linux Mint machine (Ubuntu xenial repos).. Even with the downloading instruction of ubuntugis repositories the latest version of SAGA is 2.7.
With your PR means that I'll be not able to use it anymore.. 
Should I compile SAGA LTR by myself? Are we sure that other user can easily have the SAGA LTR version installed on their machines?

Thanks!

**Comment by pcav** [comment]
Probably better to convince packagers to concentrate on saga ltr

**Comment by gioman** [comment]
> I'm on a Linux Mint machine (Ubuntu xenial repos).. Even with the downloading instruction of ubuntugis repositories the latest version of SAGA is 2.7.
> With your PR means that I'll be not able to use it anymore..
> Should I compile SAGA LTR by myself? Are we sure that other user can easily have the SAGA LTR version installed on their machines?

Hi @ghtmtt ! we should be honest and also look at the big picture: try supporting all the SAGA versions on Linux (laster year at some point we had almost one per month) was waste of time, and when they broke the compatibility of the io_gdal module it was enough for me. Fortunately they have since embraced the LTR philosophy, and seems very obvious to stick to that in all platforms. Is also planned to support other versions with a plugin. Hope that makes sense.

**Comment by ghtmtt** [comment]
@gioman thanks for sharing your ideas. Maintaining all SAGA version is a pain, really a pain, I totally agree. I also totally agree with all your ideas and moving to a SAGA LTR. I'm just a little bit scared that users (at least Ubuntu & co.) have to compile SAGA by themselves because right now it is not in the main repo. 
If SAGA LTR will be in the main packages would be great. 
Maybe we have to push in that direction..

File: python/plugins/processing/algs/saga/SagaAlgorithmProvider.py [hunk 1]
```diff
@@ -91,34 +81,28 @@ def _loadAlgorithms(self):
...
-        if version not in self.supportedVersions:
-            lastVersion = sorted(self.supportedVersions.keys())[-1]
-            if version > lastVersion:
-                version = lastVersion
-            else:
-                ProcessingLog.addToLog(ProcessingLog.LOG_ERROR,
-                                       self.tr('Problem with SAGA installation: installed SAGA version (%s) is not supported' % version))
-                return
+
+        if version not in ['2.3.0', '2.3.1']:
```
Comment (saberraz) [inline python/plugins/processing/algs/saga/SagaAlgorithmProvider.py]: the current lts is 2.3.2:
https://sourceforge.net/p/saga-gis/code/ci/release-2-3-lts/tree/saga-gis/

Comment (gioman) [inline python/plugins/processing/algs/saga/SagaAlgorithmProvider.py]: @saberraz can't seen any binaries or changelog yet https://sourceforge.net/p/saga-gis/wiki/SAGA%20Release%20Information/

Comment (johanvdw) [inline python/plugins/processing/algs/saga/SagaAlgorithmProvider.py]: There has not yet been a 2.3.2 release. Feel free to bug me with any questions you have as I'm going to maintain this branch.

**Comment by johanvdw** [comment]
@ghtmtt : I was really convinced that ubuntugis was already using 2.3.1 as it is in debian unstable already. I'll update shortly.

**Comment by ghtmtt** [comment]
@johanvdw this would be really awesome! I'm compiling 2.3 right now, but if the LTR version would be directly on the packages it would be great. Thanks!

File: python/plugins/processing/algs/saga/SagaAlgorithmProvider.py (hunk 1 above)
Comment (gioman) [inline python/plugins/processing/algs/saga/SagaAlgorithmProvider.py]: @johanvdw Many thanks! just one for now: there will be a changelog for 2.3.1 as for other releases? https://sourceforge.net/p/saga-gis/wiki/SAGA%20Release%20Information/

Comment (saberraz) [inline python/plugins/processing/algs/saga/SagaAlgorithmProvider.py]: @gioman indeed there is no release. But if you compile SAGA LTR from source, it appears as 2.3.2 and hence not picked up any more by the processing.

Comment (gioman) [inline python/plugins/processing/algs/saga/SagaAlgorithmProvider.py]: @saberraz ok good to know. I haven't yet updated the definition files anyway ;)

**Comment by saberraz** [comment]
@ghtmtt see my above comment if you are compiling the LTS/LTR from source.

File: python/plugins/processing/algs/saga/SagaAlgorithmProvider.py (hunk 1 above)
Comment (ghtmtt) [inline python/plugins/processing/algs/saga/SagaAlgorithmProvider.py]: confirmed what reported by @saberraz 
fresh compilation of SAGA 2.3.1, working fine as standalone. 
Processing *read* SAGA and lists all the algorithms, but when trying to run any of the algorith, this is the error:

```
Traceback (most recent call last):
  File "/home/matteo/lavori/QGIS/build-qgis3/output/python/plugins/processing/gui/ProcessingToolbox.py", line 257, in executeAlgorithm
    alg = alg.getCopy()
  File "/home/matteo/lavori/QGIS/build-qgis3/output/python/plugins/processing/algs/saga/SagaAlgorithm.py", line 78, in getCopy
    newone = SagaAlgorithm212(self.descriptionFile)
NameError: name 'SagaAlgorithm212' is not defined
```

**Comment by ghtmtt** [comment]
@saberraz too late :) 
anyway the SAGA standalone About says: Version 2.3.1

File: python/plugins/processing/algs/saga/SagaAlgorithmProvider.py (hunk 1 above)
Comment (saberraz) [inline python/plugins/processing/algs/saga/SagaAlgorithmProvider.py]: @ghtmtt I compiled it without gui. and this from terminal:
`saga_cmd --version`

reports:
`SAGA Version: 2.3.2`

Comment (ghtmtt) [inline python/plugins/processing/algs/saga/SagaAlgorithmProvider.py]: weird. I get:
```
saga_cmd --version
SAGA Version: 2.3.1
```

Comment (saberraz) [inline python/plugins/processing/algs/saga/SagaAlgorithmProvider.py]: here is how I got mine working with QGIS 2.18 and master:

git clone -b release-2-3-lts git://git.code.sf.net/p/saga-gis/code

For config:
./configure --disable-gui

Also, in this file, I added 2.3.2 before compiling QGISs

**Comment by johanvdw** [comment]
A  saga package for saga 2.3.1 was uploaded to ubuntugis unstable. Note that if you compile 2.3.1 from source you may run into issues with projected files if compiled with gcc6. A patch is applied in the package and in git. I should push out a 2.3.2 release with this and some other small fixes soon.

File: python/plugins/processing/algs/saga/SagaAlgorithmProvider.py (hunk 1 above)
Comment (ghtmtt) [inline python/plugins/processing/algs/saga/SagaAlgorithmProvider.py]: @saberraz @gioman so the latest LTR release on the repository is 2.3.1 that is currently not supported by Processing right?

Comment (gioman) [inline python/plugins/processing/algs/saga/SagaAlgorithmProvider.py]: @ghtmtt no yet, but I have the update of definition files in the pipeline.

Comment (saberraz) [inline python/plugins/processing/algs/saga/SagaAlgorithmProvider.py]: @ghtmtt I think there is a bit of a mixed up in the saga release versions. The branches in the saga repo are:
master
release-2-3-0
release-2-3-1
release-2-3-lts (the branch which gives you 2.3.2)
release-3.0.0 (hmmm...not consistent naming...dots and dashes!)

I think this line should change to support 2.3.x and not 2.3.0, 2.3.1 only.

Comment (gioman) [inline python/plugins/processing/algs/saga/SagaAlgorithmProvider.py]: @ghtmtt @saberraz @johanvdw I'm confused then... is then 2.3.2 the version we will have to ask to be packaged (i.e. osgeo4w and macos) and that is expected in Linux repos?

**Comment by johanvdw** [comment]
>
> @ghtmtt <https://github.com/ghtmtt> @saberraz
> <https://github.com/saberraz> @johanvdw <https://github.com/johanvdw> I'm
> confused then... is then 2.3.2 the version we will have to ask to be
> packaged (i.e. osgeo4w and macos) and that is expected in Linux repos?
>
>
For windows 2.3.1 is ok. For mac osx / linux 2.3.2 is recommended as it
contains a fix for gcc6. Debian and ubuntu already include this patch in
their 2.3.1 release (which I maintain as well).

Note that there are no interface changes between 2.3.0/2.3.1/2.3.2.

**Pull request merged**